### DISCUSSION TITLE
Jacobian Analyzer Error

### DISCUSSION THREAD

**Discussion by pj131611** [idaholab/moose#28520] [Q&A Tools] [answered]
Greetings!

I just build and compiled Moose on a new laptop and everything works out well untill I tried "analyzejacobian.py". As far as I remember, all I need to do is "analyzejacobian.py file.i", then it will automatically output a json file that contains the jacobian. But it failed. The best I can get is when I do:

`analyzejacobian.py -e main-opt -i main.i`

and I got:

`Error executing moose based application to gather DOF map`.

I also tried main-opt -i main.i --ksp_view_mat with solve_type = NEWTON, and I got

> WARNING! There are options you set that were not used!
> WARNING! could be spelling mistake, etc!
> There is one unused database option. It is:
> Option left: name:--ksp_view_mat (no value) source: command line

So I'm pretty stuck at this point. Can anyone help me out? Thanks in advance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

In main.i you can add
```
[Executioner]
  petsc_options = '-ksp_view_mat'
[]
```
you might also be able to pass `-ksp_view_mat` on the command line, with a single dash

**Reply by pj131611** [reply]
I see, thanks! What about residuals vector?

**Reply by GiudGiud** [reply]
You can print those with either a DebugResidualAux auxkernel 
Or there is an option in the Debug block iirc
